google/cloud/storage/internal/default_client.h

```diff
@@ -25,15 +25,18 @@ namespace internal {
 template <typename HttpRequestor = CurlRequest>
 class DefaultClient : public Client {
  public:
-  explicit DefaultClient(std::shared_ptr<storage::Credentials> credentials)
-      : credentials_(std::move(credentials)) {}
+  explicit DefaultClient(std::shared_ptr<Credentials> credentials)
+      : DefaultClient(ClientOptions(std::move(credentials))) {}
+
+  explicit DefaultClient(ClientOptions options)
+      : options_(std::move(options)) {}
 
   std::pair<Status, BucketMetadata> GetBucketMetadata(
       std::string const& bucket_name) override {
     // Assume the bucket name is validated by the caller.
-    HttpRequestor request("https://www.googleapis.com/storage/v1/b/" +
-                          bucket_name);
-    request.AddHeader("Authorization", credentials_->AuthorizationHeader());
+    HttpRequestor request(options_.endpoint() + "/storage/v1/b/" + bucket_name);
```

Comment:
Assuming you add the `version_` member, "v1" here should be replaced by `options_.version()`.

Reply:
Done.